czkawka_core/src/same_music.rs

```diff
@@ -229,97 +235,133 @@ impl SameMusic {
+                        Err(e) => {
+                            warnings.push(format!("Cannot open dir {}, reason {}", current_folder.display(), e));
+                            return (dir_result, warnings, fe_result);
                         }
+                    };
+
+                    // Check every sub folder/file/link etc.
+                    'dir: for entry in read_dir {
+                        let entry_data = match entry {
+                            Ok(t) => t,
+                            Err(e) => {
+                                warnings.push(format!("Cannot read entry in dir {}, reason {}", current_folder.display(), e));
+                                continue 'dir;
+                            }
+                        };
+                        let metadata: Metadata = match entry_data.metadata() {
+                            Ok(t) => t,
+                            Err(e) => {
+                                warnings.push(format!("Cannot read metadata in dir {}, reason {}", current_folder.display(), e));
+                                continue 'dir;
+                            }
+                        };
+                        if metadata.is_dir() {
+                            if !self.recursive_search {
+                                continue 'dir;
+                            }
+
+                            let next_folder = current_folder.join(entry_data.file_name());
+                            if self.directories.is_excluded(&next_folder) {
+                                continue 'dir;
+                            }
+
+                            if self.excluded_items.is_excluded(&next_folder) {
+                                continue 'dir;
+                            }
+
+                            dir_result.push(next_folder);
+                        } else if metadata.is_file() {
+                            atomic_file_counter.fetch_add(1, Ordering::Relaxed);
+
+                            let file_name_lowercase: String = match entry_data.file_name().into_string() {
+                                Ok(t) => t,
+                                Err(_inspected) => {
+                                    println!("File {:?} has not valid UTF-8 name", entry_data);
+                                    continue 'dir;
+                                }
+                            }
+                            .to_lowercase();
 
-                        let allowed_extensions = [".mp3", ".flac", ".m4a"];
+                            if !self.allowed_extensions.matches_filename(&file_name_lowercase) {
+                                continue 'dir;
+                            }
 
-                        if !allowed_extensions.iter().any(|r| current_file_name.to_string_lossy().ends_with(r)) {
-                            continue 'dir;
-                        }
+                            if ![".mp3", ".flac", ".m4a"].iter().any(|r| file_name_lowercase.ends_with(r)) {
```

Comment:
could you implement an intersection of `allowed_extensions` set with set `[".mp3", ".flac", ".m4a"]`? Thus replacing two checks with a check against set intersection.